packages/core/src/app/Module.ts

```diff
@@ -37,7 +38,7 @@ interface MenuWithSubMenus extends MenuBase {
 }
 
 interface MenuWithScreen extends MenuBase {
-  screen: string;
+  screen?: string;
```

Comment:
No, that's not logical to have a type name MenuWithScreen and to mark screen as optionnal. You need to create a new type MenuSeparator which should contains only the props taken into account in your separator and then Menu type should be  MenuWithSubMenus | RootMenuWithScreen | MenuSeparator.
Try to factorize the type to avoid duplicated code.
I think separator should take title, hideIf, parent, order and separator props into account